File: c#/src/Docmost.Sdk/Services/SSOApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class SSOApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> GetSsoProvidersAsync(PaginationOptions request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("sso/providers", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetSsoProviderAsync(SsoProviderIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("sso/info", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> CreateSsoProviderAsync(CreateSsoProviderDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("sso/create", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdateSsoProviderAsync(UpdateSsoProviderDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("sso/update", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> DeleteSsoProviderAsync(SsoProviderIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("sso/delete", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> LdapLoginAsync(LdapLoginDto request, string providerId, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("sso/ldap/" + Uri.EscapeDataString(providerId) + "/login", request, cancellationToken).ConfigureAwait(false);
    }

}

File: typescript/src/services/sso-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { CreateSsoProviderDto, LdapLoginDto, PaginationOptions, SsoProviderIdDto, UpdateSsoProviderDto } from '../models/index.js';


export class SSOApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async getSsoProviders(request: PaginationOptions): Promise<ApiResponse<unknown>> {
    return this.http.post(`sso/providers`, request);
  }

  async getSsoProvider(request: SsoProviderIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`sso/info`, request);
  }

  async createSsoProvider(request: CreateSsoProviderDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`sso/create`, request);
  }

  async updateSsoProvider(request: UpdateSsoProviderDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`sso/update`, request);
  }

  async deleteSsoProvider(request: SsoProviderIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`sso/delete`, request);
  }

  async ldapLogin(request: LdapLoginDto, providerId: string): Promise<ApiResponse<unknown>> {
    return this.http.post(`sso/ldap/${encodeURIComponent(providerId)}/login`, request);
  }

}
